content/blog/2018-11-11-colocr.md

```diff
@@ -0,0 +1,106 @@
+
+A recent publication from our laboratory included microscopy images and quantification graph. The standard analysis protocol is to use [ImageJ](https://imagej.nih.gov/ij/index.html) with a specialized plug-in for the co-localization analysis such as [coloc2](https://imagej.net/Coloc_2) or [Fiji](https://fiji.sc). The images were processed one at a time; subtracting the background, selecting regions of interest and calculating the co-localization statistics. Luckily for me, I didn’t do the work myself, but it took a good chunk of my colleague precious time. Even worse, she had to repeat this whole process more than once with minor changes in the parameters each time. At the end, there was no way to make this process reproducible, and we included only the final output in an otherwise fully reproducible article.
+
+For those two specific reasons; the need for manual work and the lack of reproducibility, I thought it would be great if I could do this analysis in [R](https://cran.r-project.org). There are several [R](https://cran.r-project.org) packages for image processing such as `EBImage`, `imager` and `magick`. These packages provided the functionality needed for conducting this analysis. So I put the pieces together in an [R](https://cran.r-project.org) package to simplify this analysis for my future self and for others who might find it useful. This is `colocr`, a simple [R](https://cran.r-project.org) package for conducting co-localization analysis of fluorescence microscopy images. In particular, it semi-automates the selections of regions of interest-arguably the most time-consuming step- and comes with a shiny app for visually comparing the outputs of different parameters before writing a final script. 
+
+## How would've we done it differently?
+
+In this recent article I referred to earlier, we assessed the co-localization of several proteins with RKIP (Raf-kinas inhibitor protein) in DU145 prostate cancer cell line. For each of these proteins, we generated multiple frames, a hundred in total. We then processed the images one by one and repeated the process with each minor change in the parameters. Now we can use `colocr`, in just a few lines of code, to apply the same analysis.
+
+A selection of the images were presented in the article in a similar figure.
+
+![](/img/blog-images/2018-11-11-colocr/images.png)
+
+
+We can calculate for the [**Pearson's** Correlation Coefficients](https://en.wikipedia.org/wiki/Pearson_correlation_coefficient) (PCC) of multiple images at once using the following chunk of code.
+
+```r
+## load required libraries
+library(tidyverse)
+library(colocr)
+
+# load images
+fls <- list.files('images', pattern = '*.jpg', recursive = TRUE)
+
+# apply co-localization test
+tst <- image_load(fls) %>%
+  roi_select(threshold = 90, n = 3) %>%
+  roi_test()
+```
+
+And the results can be visualized in a graph.
```

Comment:
Can you provide some text explanation of the plot? 
I'm having a hard time figuring out exactly what the correlations are and why for each marker labelled on the x-axis there is some spread along the x-axis.

Reply:
Might help the reader's interest if you note what the plot allowed you to conclude about the co-localization.

Reply:
Added text explanation and conclusion.
The spread on the x-axis is just jitter produced by `ggplot2::geom_jitter()`. Otherwise, all data points would line and overlap.